Document/0x05b-Basic-Security_Testing.md

```diff
@@ -408,46 +408,51 @@ Other resources where you might find useful information are:
+- Scan process memory for occurrences of a string ([Memory API](https://www.frida.re/docs/javascript-api/#memory "Frida - Memory API")).
+- Intercept native function calls to run your own code at function entry and exit ([Interceptor API](https://www.frida.re/docs/javascript-api/#interceptor "Frida - Interceptor API")).
 
-![Frida](Images/Chapters/0x04/frida.png)
+Remember that on Android, you can also benefit from the built-in tools provided when installing Frida, that includes the Frida CLI, frida-ps, frida-ls-devices and frida-trace, to name some of them.
 
-*FRIDA Architecture, source: [https://www.frida.re/docs/hacking/](https://www.frida.re/docs/hacking)*
+Frida is often compared to Xposed, however this comparison is far from fair as both frameworks were designed with different goals in mind. This is important to understand as an app security tester so that you can know which framework to use in which situation:
 
-Here are some more APIs FRIDA offers on Android:
+- Frida is standalone, all you need is to run the frida-server binary from a known location in your target Android device (see "Installing Frida" below). This means that, in contrast to Xposed, it is not _deep_ installed in the target OS.
+- Reversing an app is an iterative process. As a consequence of the previous point, you obtain a shorter feedback loop when testing as you don't need to (soft) reboot to apply or simply update your hooks. So you might prefer to use Xposed when implementing more permanent hooks.
+- You may inject and update your Frida JavaScript code on the fly at any point during the runtime of your process (similarly to Cycript on iOS). This way you can perform the so-called _early instrumentation_ by letting Frida spawn your app or you may prefer to attach to a running app that you might have brought to a certain state.
+- Frida is able to handle both Java as well as native code (JNI), allowing you to modify both of them. This is unfortunately a limitation of Xposed which lacks of native code support.
 
-- Instantiate Java objects and call static and non-static class methods
-- Replace Java method implementations
-- Enumerate live instances of specific classes by scanning the Java heap (Dalvik only)
-- Scan process memory for occurrences of a string
-- Intercept native function calls to run your own code at function entry and exit
-
-The FRIDA Stalker —a code tracing engine based on dynamic recompilation— is available for Android (with support for ARM64), including various enhancements, since Frida version 10.5 ([https://www.frida.re/news/2017/08/25/frida-10-5-released/](https://www.frida.re/news/2017/08/25/frida-10-5-released/)). Some features have limited support on current Android devices, such as support for ART ([https://www.frida.re/docs/android/](https://www.frida.re/docs/android/)), so it is recommended to start out with the Dalvik runtime.
```

Comment:
no more frida-stalker?

Reply:
The sentence is not really useful as-is: it's informing us about its existence "with enhancements" and that is for Android, actually is a general tool. We're not using it yet in the guide, so I thought I'd write a separate ticket which should:

- find a good place for introducing stalker (probably in the Frida general intro in 0x4c)
- give a really useful description and introduction.

Reply:
good idea : let's move it to another ticket.